manager/controllers/app/data_catalog_interface.go

```diff
@@ -16,7 +19,44 @@ import (
 // Returns:
 // - an error if happened
 // - the new asset identifier
-func (r *FybrikApplicationReconciler) RegisterAsset(catalogID string, info *app.DatasetDetails,
+func (r *FybrikApplicationReconciler) RegisterAsset(assetID string, catalogID string, info *app.DatasetDetails,
 	input *app.FybrikApplication) (string, error) {
-	return "", errors.New("unsupported feature")
+	r.Log.Trace().Msg("RegisterAsset")
+	details := datacatalog.ResourceDetails{}
+	if info.Details != nil {
+		infoDetails := info.Details
```

Comment:
No need to copy the structure, just use info.Details.<field>

Reply:
Fixed. Thanks